### ISSUE TITLE
'Custom element doesn't exist' error in HA mobile app

### ISSUE THREAD

**Issue by rfehr613** [dwainscheeren/dwains-lovelace-dashboard#707]
I installed Dwain's dashboard last night, and I get the 'custom element doesn't exist' error shown in your installation steps. I cleared the app and browser cache as stated, but it still isn't working in the HA app (it works in my browser though). I also use duckdns, but i don't see any way to clear the cache for that.  I also don't think that would matter, as i use duckdns for access remotely as well as locally and in both browser and HA app. Seems to be a problem with only the app.

**Comment by teamadi1** [comment]
This is an issue I have also started to consistently see recently

**Comment by cmariegaard** [comment]
This issue is also mentioned in #691 
DD3.x Issue is persistent HA Companion App after latest HA/OS updates.
The quickfix to clear cache only works one time, after closing Companion App and reopen the issue is back.
No problem in DD3.x  in PC browser

**Comment by teamadi1** [comment]
In relation to the fix suggested on another ticket to reinstall the companion app - I have done this, and the issue has returned

**Comment by iroQuai** [comment]
Man this error keepa returning, clearing cache doesn't help most of the time. Will reinstall again and hopefully this wil fix it (temporary?)

**Comment by ghost** [comment]
This might be the problem.
The only mobile or tablet that gets this problem is my pixel. It seems like the problem occurs if I do a release upgrade from the mobile. All other devices reloads after the upgrade except for the one that done the upgrade.
Got it 2023.7 and it magically disappeared after an upgrade after that, that I probably did from my laptop.
Got the problem on my pixel again now with 2023.9 and I know I did that from the pixels android app.
All other apps, Apple and Android works fine.
My hunch is that the frontend isn't reloaded/cleared properly after the upgrade process is finished. The other devices notices at startup that there is a discrepancy between app version of files and server version and reloads them properly.
This is only a working theory atm, I will continue to test

**Comment by rosicenko** [comment]
@ekalle-swe can confirm same issue. Also Pixel user (7 Pro) and after core update tu 2023.9 HAAS app stopped working. Nothing helps or only for a few secs.

**Comment by iroQuai** [comment]
Same here, although not so consistent. Samsung s10 here

Op do 7 sep. 2023 15:18 schreef Rostislav Zlatnik ***@***.***

**Comment by ghost** [comment]
> @ekalle-swe can confirm same issue. Also Pixel user (7 Pro) and after core update tu 2023.9 HAAS app stopped working. Nothing helps or only for a few secs.

So you doesn't misunderstand.
The issue is not Pixel, the issue is doing an upgrade from the app if a frontend refresh is needed. It kind of uses those files when you run from the app and I suspect it will be an incomplete refresh, as you are locking those files by using then. When you do a first startup on another device running same app, the refresh is done before the gui is started, no files are locked and it will be a successful refresh

**Comment by cmariegaard** [comment]
After lastest HA update 2023.9 - again trouble for most of my devices getting DD loaded.
Have testet on 3 devices with mobile app and 4 PC's with Chrome Browser - none of these can load DD.
Clearing cache doesn't help, on mobile devices clearing both cache & data also doesn't solve anything.
Some phones can load once after clearing, but then they fails again.

**Comment by ghost** [comment]
Now, I'm more convinced that the problem occurs only if you update haos from an app device 
I upgraded to 2023.0 from android. Got no question of "reload frontend" and the problem started. Other devices got "reload frontend" at first startup. Web also suffers but a simple F5 solves the problem 

I now updated to 2309.1 and did it from web
- web interface get the problem after update. F5 solves problem.
- android or apple without problem got "reload frontend" on startup and worked fine.
- android or apple with problem also asks "reload frontend"
on first startup after update and it solves the problem.

Much points at the problem only occurs ig you do update from app. frontend will not be automatically reloaded and you will never be asked to reload. Same problem on web but there you have the choice to do reload manually via F5 button.

The bug lies in the update process but it would be much helpful to have a "reload frontend" choice, somewhere in the settings.

To avoid

Never run update from app, only web.
You will get away with it if no update is done to frontend and no reload will then be needed.

**Comment by cmariegaard** [comment]
@ekalle-swe For my HA it's not the same device behavior, I can't even get the DD in web browser to load again with F5. Even tried to reinstall mobile app, which solved the issue last time, but no success this time.

**Comment by ghost** [comment]
Strange?

Could be different scenarios that leads to the same result, probably.
I just happened to find one of the scenarios but there could be others.
In my test, AL devices was exited at update, so the only one active when I used F5 was the web console.
I think it's wise to have everyone close their apps when you do an update.
Tried Shift+F5 as well?

**Comment by cmariegaard** [comment]
Yes I tried several cache clearing methods on browsers, and on phone app even a reinstall still giving problems.
In my setup most of the PC's was not powered on when doing the HA updates.
In thread #726 comment https://github.com/dwainscheeren/dwains-lovelace-dashboard/issues/726#issuecomment-1712777714 mentions that Firefox is working, and I can confirm that on my PC's as well.

Update: I just tried to do a manuel update to Dwains 3.5.0 test source code, and it seems to work on my HA for both Chrome/Edge browsers after a Shift+F5 and Companion App as well.
The only strange observation I see in Companion App is, when i switch to DD 3.5.0 test page, I can't open the original HA menus (burger menu), I can only go back with the phones 'back' button

**Comment by dwainscheeren** [comment]
Thanks for reporting this.

Dwains Dashboard 3.9.0 has now been released, and the source files for DD3 are available in this repository.

DD3 is no longer under active feature development and will slowly be phased out. Active development continues in Dwains Dashboard Next:
https://github.com/dwainscheeren/dwains-dashboard-next

At the time of writing, Dwains Dashboard Next is still in alpha.

If this issue is a feature request, please check whether it fits Dwains Dashboard Next instead.

If this issue is a bug, please retest with DD3 version 3.9.0. If the bug is still reproducible there, please open a new issue with:

- Home Assistant version
- Dwains Dashboard version
- browser and device
- browser console errors
- clear reproduction steps

For more context, see #906.

For that reason, this older issue is being closed as part of the DD3 issue tracker cleanup.
